x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -150,31 +172,28 @@ EXPR_MULTILINE_COMMENT
 EXPR_WS
     : WS -> channel(HIDDEN)
     ;
+//
+// FROM command
+//
+mode FROM_MODE;
+FROM_PIPE : PIPE -> type(PIPE), popMode;
+FROM_OPENING_BRACKET : OPENING_BRACKET -> type(OPENING_BRACKET), pushMode(FROM_MODE), pushMode(FROM_MODE);
+FROM_CLOSING_BRACKET : CLOSING_BRACKET -> type(CLOSING_BRACKET), popMode, popMode;
+FROM_COMMA : COMMA -> type(COMMA);
+FROM_ASSIGN : ASSIGN -> type(ASSIGN);
```

Comment:
Type reuse and consistency.

Reply:
Why do we need to push/pop `FROM_MODE` twice?

Reply:
I think it's because of the double push the opening_bracket is doing: https://github.com/elastic/elasticsearch/pull/100740/files#diff-e78771d02134981fe1563249a0fbbdabf95b40299b2086dd19fccef6e87f39c4R150
Same for closing_bracket.